simtools/corsika/corsika_config.py

```diff
@@ -326,15 +326,11 @@ def _set_primary_particle(self, args_dict):
             Primary particle.
 
         """
-        if not args_dict:
+        if not args_dict or args_dict.get("primary_id_type") is None:
             return PrimaryParticle()
```

Comment:
This just raises an error, right?

Reply:
No - `bool(None)` is `False`, this simply sets all variables to None in the constructor.